### PR TITLE
Increase upgrades test coverage

### LINKED ISSUE #4809: Test every public version of Unlock
**Is your feature request related to a problem? Please describe.**
Our upgrades test currently tests v0 -> the latest.

**Describe the solution you'd like**
We should test each public version.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by nfurfaro**
# Description
This renames a test file from `upgrades.js` to `v0ToLatest.js`, and edits a few names in the file to reflect this. It also tests upgrading each of the other versions (v1,v2,v3,v4) to the latest. (v5 test is included as well. While not yet needed, as soon as we deploy the next version it will be useful.)
I've copied & pasted the same test file for each, modifying only version numbers and, where needed, function parameters to reflect the api of the version being tested.

Note: I've added a `latestVersion.js` file so we can update these number in one place after an upgrade.

This

# Issues

Fixes #4809 
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [ ] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: smart-contracts/test/Unlock/upgrades/v0ToLatest.js [hunk 1]
```diff
@@ -65,14 +65,19 @@ contract('Unlock / upgrades', accounts => {
...
-  describe('v1', () => {
+  describe('latest', () => {
     before(async () => {
-      project.upgradeProxy(proxy.address, UnlockV1)
-      unlock = UnlockV1.at(proxy.address)
+      project.upgradeProxy(proxy.address, UnlockLatest)
+      unlock = UnlockLatest.at(proxy.address)
+    })
+
+    it('Latest Unlock version is correct', async () => {
```
Comment (nfurfaro) [inline smart-contracts/test/Unlock/upgrades/v0ToLatest.js]: This is failing with a revert:
```
1) Contract: Unlock / upgrades
       latest
         Latest Unlock version is correct:
     Error: Returned error: VM Exception while processing transaction: revert
      at Object.ErrorResponse (node_modules/@openzeppelin/upgrades/node_modules/web3-core-helpers/src/errors.js:29:16)
      at /Users/overlord/DAPPS/unlock/smart-contracts/node_modules/@openzeppelin/upgrades/node_modules/web3-core-requestmanager/src/index.js:140:36
      at /Users/overlord/DAPPS/unlock/smart-contracts/node_modules/truffle/build/webpack:/packages/provider/wrapper.js:112:1
      at XMLHttpRequest.request.onreadystatechange (node_modules/truffle/build/webpack:/~/web3-providers-http/src/index.js:96:1)
      at XMLHttpRequestEventTarget.dispatchEvent (node_modules/truffle/build/webpack:/~/xhr2-cookies/dist/xml-http-request-event-target.js:34:1)
      at XMLHttpRequest._setReadyState (node_modules/truffle/build/webpack:/~/xhr2-cookies/dist/xml-http-request.js:208:1)
      at XMLHttpRequest._onHttpResponseEnd (node_modules/truffle/build/webpack:/~/xhr2-cookies/dist/xml-http-request.js:318:1)
      at IncomingMessage.<anonymous> (node_modules/truffle/build/webpack:/~/xhr2-cookies/dist/xml-http-request.js:289:47)
      at endReadableNT (_stream_readable.js:1064:12)
      at _combinedTickCallback (internal/process/next_tick.js:139:11)
      at process._tickCallback (internal/process/next_tick.js:181:9)
```

File: smart-contracts/test/Unlock/upgrades/v0ToLatest.js [hunk 2]
```diff
@@ -65,14 +65,19 @@ contract('Unlock / upgrades', accounts => {
...
+      const unlockVersion = await unlock.methods.unlockVersion().call()
+      assert.equal(unlockVersion, 5)
```
Comment (nfurfaro) [inline smart-contracts/test/Unlock/upgrades/v0ToLatest.js]: This fails:
```
1) Contract: Unlock / upgrades
       latest
         Latest Unlock version is correct:
     Error: Returned error: VM Exception while processing transaction: revert
      at Object.ErrorResponse (node_modules/@openzeppelin/upgrades/node_modules/web3-core-helpers/src/errors.js:29:16)
      at /Users/overlord/DAPPS/unlock/smart-contracts/node_modules/@openzeppelin/upgrades/node_modules/web3-core-requestmanager/src/index.js:140:36
      at /Users/overlord/DAPPS/unlock/smart-contracts/node_modules/truffle/build/webpack:/packages/provider/wrapper.js:112:1
      at XMLHttpRequest.request.onreadystatechange (node_modules/truffle/build/webpack:/~/web3-providers-http/src/index.js:96:1)
      at XMLHttpRequestEventTarget.dispatchEvent (node_modules/truffle/build/webpack:/~/xhr2-cookies/dist/xml-http-request-event-target.js:34:1)
      at XMLHttpRequest._setReadyState (node_modules/truffle/build/webpack:/~/xhr2-cookies/dist/xml-http-request.js:208:1)
      at XMLHttpRequest._onHttpResponseEnd (node_modules/truffle/build/webpack:/~/xhr2-cookies/dist/xml-http-request.js:318:1)
      at IncomingMessage.<anonymous> (node_modules/truffle/build/webpack:/~/xhr2-cookies/dist/xml-http-request.js:289:47)
      at endReadableNT (_stream_readable.js:1064:12)
      at _combinedTickCallback (internal/process/next_tick.js:139:11)
      at process._tickCallback (internal/process/next_tick.js:181:9)
```

File: smart-contracts/test/Unlock/upgrades/v0ToLatest.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ ZWeb3.initialize(web3.currentProvider)
 const UnlockV0 = Contracts.getFromNodeModules('unlock-abi-0', '../../Unlock')
 const PublicLockV0 = require('public-lock-abi-0/abi_V0')
 
-const UnlockV1 = Contracts.getFromLocal('Unlock')
+const UnlockLatest = Contracts.getFromLocal('Unlock')
```
Comment (nfurfaro) [inline smart-contracts/test/Unlock/upgrades/v0ToLatest.js]: `getFromLocal` should always be the latest version of Unlock, correct?

Comment (nfurfaro) [reply smart-contracts/test/Unlock/upgrades/v0ToLatest.js]: I should probably modify the naming for publicLock as well to reflect the fact that we're using the latest, not necessarily v1.

File: smart-contracts/package.json [hunk 4]
```diff
@@ -33,6 +33,7 @@
     "unlock-abi-0-1": "1.0.0",
     "unlock-abi-0-2": "1.0.0",
     "unlock-abi-1-0": "1.0.0",
+    "unlock-abi-1-1": "^1.0.0",
```
Comment (nfurfaro) [inline smart-contracts/package.json]: I could do this in a separate pr and then rebase. This was already in `node_modules` which is why I missed it, but CI is complaining that it needs to be in our `package.json`.

Comment (NickCuso) [reply smart-contracts/package.json]: Should pin that as well (i.e. remove the `^`)

Comment (nfurfaro) [reply smart-contracts/package.json]: Right. I've just done this in a separate PR.

**Comment by nfurfaro** [comment]
Agreed it's pretty verbose, and yes, this could be a good refactoring project.
Re: 

> That may have the added benefit of highlighting breaking API changes.

I think either way breaking api changes will show up here (at least changes to the functions used in these tests).

**Pull request merged**